tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs

```diff
@@ -221,5 +223,40 @@ public void TestConstructor_WithCertificate_X509Certificate2()
 
             Assert.IsNotNull(cca.AppConfig.ClientCredentialCertificate);
         }
+
+        [TestMethod]
+        [DeploymentItem(@"Resources\CustomInstanceMetadata.json")]
+        public void TestConstructor_WithValidInstanceDicoveryMetadata()
+        {
+            string instanceMetadataJson = File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("CustomInstanceMetadata.json"));
+            var cca = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                   .WithInstanceDicoveryMetadata(instanceMetadataJson)
+                                                   .Build();
+
+            var instanceDiscoveryMetadata = (cca.AppConfig as ApplicationConfiguration).CustomInstanceDiscoveryMetadata;
+            Assert.AreEqual(2, instanceDiscoveryMetadata.Metadata.Length);
+        }
+
+        [TestMethod]
+        [DeploymentItem(@"Resources\CustomInstanceMetadata.json")]
+        public void TestConstructor_InstanceMetadata_ValidateAuthority_MutuallyExclusive()
+        {
+            string instanceMetadataJson = File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("CustomInstanceMetadata.json"));
+            var ex = AssertException.Throws<MsalClientException>(() => ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                  .WithInstanceDicoveryMetadata(instanceMetadataJson)
+                                                  .WithAuthority("https://some.authority/bogus/", true)
+                                                  .Build());
+            Assert.AreEqual(ex.ErrorCode, MsalError.ValidateAuthorityOrCustomMetadata);
+        }
+
+        [TestMethod]
+        public void TestConstructor_BadInstanceMetadata()
+        {
+            var ex = AssertException.Throws<MsalClientException>(() => ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                  .WithInstanceDicoveryMetadata("{bad_json_metadata")
```

Comment:
>{ [](start = 81, length = 1)

nit: ?

Reply:
We need a test that checks if the user configures a bad json. This is it.